src/auto-evo/steps/GenerateMiche.cs

```diff
@@ -69,17 +69,31 @@ public Miche GenerateMicheTree(AutoEvoGlobalCache globalCache)
             generatedMiche.AddChild(ironMiche);
         }
 
+        var hasSmallSulfurChunk =
+            patch.Biome.Chunks.TryGetValue("sulfurSmallChunk", out var smallSulfurChunk) &&
+            smallSulfurChunk.Density > 0;
+
+        var hasMediumSulfurChunk = patch.Biome.Chunks.TryGetValue("sulfurMediumChunk", out var mediumSulfurChunk) &&
+            mediumSulfurChunk.Density > 0;
+
         // Hydrogen Sulfide
-        if (patch.Biome.TryGetCompound(Compound.Hydrogensulfide, CompoundAmountType.Biome,
-                out var hydrogenSulfideAmount) &&
-            hydrogenSulfideAmount.Amount > 0)
+        if ((patch.Biome.TryGetCompound(Compound.Hydrogensulfide, CompoundAmountType.Biome,
+                    out var hydrogenSulfideAmount) &&
+                hydrogenSulfideAmount.Amount > 0) || hasSmallSulfurChunk || hasMediumSulfurChunk)
         {
             var hydrogenSulfideMiche = new Miche(globalCache.HydrogenSulfideConversionEfficiencyPressure);
             var generateATP = new Miche(globalCache.MinorGlucoseConversionEfficiencyPressure);
             var maintainGlucose = new Miche(globalCache.MaintainGlucose);
-            var envPressure = new Miche(globalCache.HydrogenSulfideCloudPressure);
 
-            maintainGlucose.AddChild(envPressure);
+            if (hydrogenSulfideAmount.Amount > 0)
```

Comment:
If hydrogenSulfide is not present in the patch, just the chunks, I might be wrong, but isn't this a crash?

Reply:
I think my IDE or the C# compiler should have prevented that. As hydrogenSulfideAmount is a struct, when the TryGet fails it returns a default initialized struct instance (i.e. so all values are initialized to 0, which should mean the amount is zero in the case that the TryGet failed). And that should mean writing this check is safe like this.